matrixprofile/matrixProfile.py

```diff
@@ -122,6 +179,9 @@ def _matrixProfile_sampling(tsA,m,orderClass,distanceProfileFunction,tsB=None,sa
     order = orderClass(len(tsA)-m+1, random_state=random_state)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
 
```

Comment:
All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB.

```python
if not is_array_like(tsB):
    tsB = tsA
```